### ISSUE TITLE
CCDM: history actions do not work as in v14

### ISSUE THREAD

**Issue by manolo** [vaadin/flow#7572]
Enable client-side mode in `flow-tests/test-root-context` and run either `HistoryIT` `BackButtonServerRoundTripIT`, they will fail

```
mvn verify -Dvaadin.useDeprecatedV14Bootstrapping=false -Dit.test=HistoryIT,BackButtonServerRoundTripIT
```

**Comment by ajgassner** [comment]
Currently we are unable to use Vaadin 15 in our project. Everything works fine except the history API and router links. We aren't using client side views yet.

E.g. `UI.getCurrent().page.history.back()` or `RouterLink("test", SomeView::class.java)` are unreliable, they change the URL but no content changes on the page sometimes. Same behavior tested directly in the browser's console with `history.back()`. 

I suggest to to raise the severity of this issue, it's a real problem. I think it's somehow caused by vaadin-router on the client side.

**Comment by haijian-vaadin** [comment]
Hi @ajgassner thanks a lot for the feedback. It indeed sounds more severe than we originally estimated. Do you have a project that we can use to reproduce the problem?

**Comment by ajgassner** [comment]
Mhm i just tried to reproduce the issue with a small sample application. With this minimal example it works as expected. I really don't know what's causing the problem in our larger production application unfortunately. There's also nothing special or custom.

**Comment by haijian-vaadin** [comment]
Hi @ajgassner, do you happen to use `ui.navigate()`? wonder if https://github.com/vaadin/flow/issues/8071 is related or not.

**Comment by ajgassner** [comment]
Hi @haijian-vaadin!

Yes that‘s the case.

**Comment by haijian-vaadin** [comment]
Thanks for the demo @ajgassner, wow, you are using Kotlin, cool!

**Comment by haijian-vaadin** [comment]
@ajgassner this issue should now be fixed in the @vaadin/router 1.7.2.
You can try to delete the `node_modules` folder and the `package-lock.json` in your project, then it should be able to get @vaadin/router 1.7.2. You can give it a try and let us know if it works for you.

**Comment by haijian-vaadin** [comment]
`HistoryIT` is still failing, but keep this ticket closed and open https://github.com/vaadin/flow/issues/8213 for tracking this particular issue.
